### PR TITLE
Feature/uploading files

### PR DESCRIPTION
**Pull request by askirk**
Unsure whether I'm right about using PutFile with pre-existing filename overwrites existing file.

### CODE REVIEW

File: docs/Uploading Files and Graphics/index.md [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+putFileRequest.setPersistentFile(true);
+putFileRequest.setFileData(file_data);
+putFileRequest.setCorrelationID(CorrelationIdGenerator.generateId());
+putFileRequest.setOnRPCResponseListener(new OnRPCResponseListener() {
+
+    @Override
+    public void onResponse(int correlationId, RPCResponse response) {
+        setListenerType(UPDATE_LISTENER_TYPE_PUT_FILE); // necessary for PutFile requests
+
+        if(((PutFileResponse) response).getSuccess()){
```
Comment (joeygrover) [inline docs/Uploading Files and Graphics/index.md]: Unnecessary class casting. All responses should have the getSucess method

**Comment by joeygrover** [comment]
You should look at #5 and add the helper method to fill a PutFile with data from a resource file. It's very handy

File: docs/Uploading Files and Graphics/index.md (hunk 1 above)
Comment (joeygrover) [inline docs/Uploading Files and Graphics/index.md]: This goes for all responses just wanting to check its success or not

Comment (askirk) [inline docs/Uploading Files and Graphics/index.md]: Yeah, bad habit that migrated through a lot of my docs

File: docs/Uploading Files and Graphics/index.md [hunk 2]
```diff
@@ -0,0 +1,198 @@
+## Uploading Files and Graphics
+Graphics allow for you to better customize what you would like to have your users see, and provide a better User Interface.
```
Comment (joeygrover) [inline docs/Uploading Files and Graphics/index.md]: Remove comma after "...your users see"

File: docs/Uploading Files and Graphics/index.md [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+putFileRequest.setPersistentFile(true);
+```
+
+!!! note
+Be aware that persistance will not work if space on the head unit is limited.
+!!!
+
+### Overwrite Stored Files
+If a file being uploaded has the same name as an already uploaded file, the new file will overwrite the previous file. 
+
```
Comment (joeygrover) [inline docs/Uploading Files and Graphics/index.md]: Can we insert a code snippet? Should just be a single line like the file persistence

**Pull request merged**